examples_tests/0.ImportanceSamplingEnvMaps/main.cpp

```diff
@@ -0,0 +1,668 @@
+
+			if (!sum_filter.execute(std::execution::par_unseq, &state))
+				std::cout << "SAT filter failed for some reason" << std::endl;
+
+			_NBL_ALIGNED_FREE(state.scratchMemory);
+
+			// From the outImage you gotta extract integral and normalize
+			double* marginalCdfPixel = (double*)marginalCdfImage->getBuffer()->getPointer();
+
+			marginalIntegral = marginalCdfPixel[pdfDomainExtent.Y - 1];
+
+			// now normalize
+			for (uint32_t y = 0; y < pdfDomainExtent.Y; ++y)
+				marginalCdfPixel[y] /= marginalIntegral;
+		}
+		
+		for (uint32_t i = 1; i < (marginalCdfImage->getBuffer()->getSize() / sizeof(double)); ++i)
+			assert(((double*)marginalCdfImage->getBuffer()->getPointer())[i] > ((double*)marginalCdfImage->getBuffer()->getPointer())[i - 1]);
+
+		// Computing LUTs
+
+		const uint32_t phiPdfLUTChannelCount = 2u; // phi and pdf
+		const size_t phiPdfLUTBufferSize = pdfDomainExtent.X * pdfDomainExtent.Y * phiPdfLUTChannelCount * sizeof(float);
+		core::smart_refctd_ptr<ICPUBuffer> phiPdfLUTBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(phiPdfLUTBufferSize);
+		memset(phiPdfLUTBuffer->getPointer(), 0, phiPdfLUTBufferSize);
+
+		const uint32_t thetaLUTChannelCount = 1u; // theta
+		const size_t thetaLUTBufferSize = pdfDomainExtent.Y * thetaLUTChannelCount * sizeof(float);
+		core::smart_refctd_ptr<ICPUBuffer> thetaLUTBuffer = core::make_smart_refctd_ptr<ICPUBuffer>(thetaLUTBufferSize);
+		memset(thetaLUTBuffer->getPointer(), 0, thetaLUTBufferSize);
+
+		float* phiPdfLUTPixel = (float*)phiPdfLUTBuffer->getPointer();
+		float* thetaLUTPixel = (float*)thetaLUTBuffer->getPointer();
+
+		core::vector2d<double> xi(0.0, 0.0);
+		core::vector2d<double> xiRemapped = { 0.0, 0.0 };
+		for (uint32_t y = 0; y < pdfDomainExtent.Y; ++y)
+		{
+			xi.Y = (y + 0.5) / (double)pdfDomainExtent.Y;
+
+			int32_t offset = bisectionSearch((double*)marginalCdfImage->getBuffer()->getPointer(), pdfDomainExtent.Y, xi.Y, &xiRemapped.Y);
+			const uint32_t rowToSample = (uint32_t)offset + 1u;
+			double marginalPdf = ((double*)conditionalIntegrals->getPointer())[offset + 1] / marginalIntegral;
+
+			const double theta = xiRemapped.Y * core::PI<double>();
+			*thetaLUTPixel++ = (float)theta;
+
+			for (uint32_t x = 0; x < pdfDomainExtent.X; ++x)
+			{
+				xi.X = (x + 0.5) / (double)pdfDomainExtent.X;
+
+				const int32_t offset = bisectionSearch((double*)conditionalCdfImage->getBuffer()->getPointer() + rowToSample * pdfDomainExtent.X, pdfDomainExtent.X, xi.X, &xiRemapped.X);
+				const double conditionalPdf = ((double*)luminancePdfBuffer->getPointer())[rowToSample * pdfDomainExtent.X + offset + 1] / ((double*)conditionalIntegrals->getPointer())[rowToSample];
+
+				const double phi = xiRemapped.X * 2.0 * core::PI<double>();
+				const double pdf = (core::sin(theta) == 0.0) ? 0.0 : (marginalPdf * conditionalPdf) / (2.0 * core::PI<double>() * core::PI<double>() * core::sin(theta));
+
+				*phiPdfLUTPixel++ = (float)phi;
+				*phiPdfLUTPixel++ = (float)pdf;
+			}
+		}
```

Comment:
turn this into a simple, new image filter called `Inverse2DConditionalCDFFilter`, inherit from `CMatchedSizeInOutImageFilterCommon`, add extra image parameters in the state for the `inMarginalCDF` and `outMarginalInvCDF` (use the already exisitng `CMatchedSizeInOutImageFilterCommon` in and out images as the conditional 2D CDF and the inverse 2D CDF)

Kinda important cause I want to reuse a lot of the code here for example 22